app/views/cart_js/update_cart.js.erb

```diff
@@ -1,4 +1,17 @@
 $('#list_items_in_cart').replaceWith('<%= escape_javascript(render :partial => 'reservations/list_items_in_cart') %>');
+<% # availability updates %>
+<% if @ems %>
+  <% @ems.each do |em| %>
+    $('<%= "#availability_#{em}" %>').html(<%= @availability_hash[em] %>);
+    <% if @availability_hash[em] <= 0 %>
+      $('<%= "#add_to_cart_#{em}" %>').children('.btn').addClass('disabled', 'add_to_cart_disabled');
+      $('<%= "#add_to_cart_#{em}" %>').children('.btn').removeClass('add_to_cart');
```

Comment:
Ha I can't believe I did that. I think I added them in separate commits, still makes no sense :-P. I'll also store the query in a variable and rebase.